dhis-2/dhis-api/src/main/java/org/hisp/dhis/user/SystemUser.java

```diff
@@ -0,0 +1,169 @@
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.user;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.hisp.dhis.common.CodeGenerator;
+import org.hisp.dhis.security.Authorities;
+import org.springframework.security.core.GrantedAuthority;
+
+/**
+ * @author Morten Svanæs <msvanaes@dhis2.org>
+ */
+public class SystemUser implements UserDetails {
+
+  @Override
+  public Collection<? extends GrantedAuthority> getAuthorities() {
+    return List.of((GrantedAuthority) Authorities.ALL::name);
+  }
+
+  @Override
+  public Set<String> getAllAuthorities() {
+    return Set.of(Authorities.ALL.name());
+  }
+
+  @Override
+  public String getPassword() {
+    return null;
+  }
+
+  @Override
+  public String getUsername() {
+    // Never rely on this method to get the username of the system user for identification.
+    return "system-process_" + CodeGenerator.generateUid();
+  }
+
+  @Override
+  public boolean isAccountNonExpired() {
+    return true;
+  }
+
+  @Override
+  public boolean isAccountNonLocked() {
+    return true;
+  }
+
+  @Override
+  public boolean isCredentialsNonExpired() {
+    return true;
+  }
+
+  @Override
+  public boolean isEnabled() {
+    return true;
+  }
+
+  @Override
+  public boolean isSuper() {
+    return true;
+  }
+
+  @Override
+  public String getUid() {
+    return "system-process";
+  }
+
+  @Override
+  public Long getId() {
+    return -1L;
+  }
+
+  @Override
+  public String getCode() {
+    return "code_" + CodeGenerator.generateUid();
+  }
+
+  @Override
+  public String getFirstName() {
+    return "system";
+  }
+
+  @Override
+  public String getSurname() {
+    return "user";
+  }
+
+  @Override
+  public Set<String> getUserGroupIds() {
+    return Set.of();
+  }
+
+  @Override
+  public Set<String> getUserOrgUnitIds() {
+    return Set.of();
+  }
+
+  @Override
+  public boolean hasAnyAuthority(Collection<String> auths) {
+    return false;
```

Comment:
should this be true?
if this method's purpose is to check if a user has a specific auth from a set of auths (which i imagine is to decide whether access is granted on something or not) then should this be true? (ALL auth trumps everything)